packages/test/src/test-workflows.ts

```diff
@@ -1853,7 +1894,7 @@ test('conditionRacer', async (t) => {
         makeFireTimerJob(1)
       )
     );
-    compareCompletion(t, completion, makeSuccess([{ cancelTimer: { seq: 1 } }]));
+    compareCompletion(t, completion, makeSuccess([], [SdkFlags.ProcessWorkflowActivationJobsAsSingleBatch]));
```

Comment:
Why did this cancel timer go away?

Reply:
That CancelTimer command was bogus. We just sent a `FireTimer(1)`. Why would we cancel an already fired timer?

That timer comes from the `condition(fn, timeout)` line. In this case, `fn` will return true as soon as the signal handler is invoked. Since signals were procesed in a distinct phase, and timer events in a second phase, the microtasks created internally by `condition()` would have the opportunity to settle on a state where "`fn` returns true, because the signal has been received, but timer has not yet fired, because the job has not yet been processed".

`ProcessWorkflowActivationJobsAsSingleBatch` fixes that. As soon as the `fn` returns true, the internal condition promise (the one without a timer) gets resolved, but by the time the `then` microtask gets executed, the `TimerFired` event has also been processed, and so the attempt to cancel the timer is known to be a noop.

Just for extra safety, I modified that test to return the boolean from `condition(fn, timeout)`, so we know that we didn't break the documented semantic of that API.